ui/oopsyraidsy/data/04-sb/dungeon/bardams_mettle.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+// Bardam's Mettle
+[{
+  zoneRegex: {
+    en: /^Bardam's Mettle$/,
+    cn: /^巴儿达木霸道$/,
+    ko: /^바르담 패도$/,
+  },
+  zoneId: ZoneId.BardamsMettle,
+  damageWarn: {
+    'Bardam Dirty Claw': '21A8', // Frontal cleave, Gulo Gulo trash
+    'Bardam Epigraph': '23AF', // Line AoE, Wall of Bardam trash
+    'Bardam The Dusk Star': '2187', // Circle AoE, environment before first boss
+    'Bardam The Dawn Star': '2186', // Circle AoE, environment before first boss
+    'Bardam Crumbling Crust': '1F13', // Circle AoEs, Garula, first boss
+    'Bardam Ram Rush': '1EFC', // Line AoEs, Steppe Yamaa, first boss.
+    'Bardam Lullaby': '24B2', // Circle AoEs, Steppe Sheep, first boss.
+    'Bardam Heave': '1EF7', // Frontal cleave, Garula, first boss
+    'Bardam Wide Blaster': '24B3', // Enormous frontal cleave, Steppe Coeurl, first boss
+    'Bardam Double Smash': '26A', // Circle AoE, Mettling Dhara trash
+    'Bardam Transonic Blast': '1262', // Circle AoE, Steppe Eagle trash
+    'Bardam Wild Horn': '2208', // Frontal cleave, Khun Gurvel trash
+    'Bardam Heavy Strike 1': '2578', // 1 of 3 270-degree ring AoEs, Bardam, second boss
```

Comment:
Just to follow up here so the thought doesn't get lost, as per offline discussion, all of the Bardam abilities are not damage (`0E` low byte on the flags, and so get ignored by `damageRegex` in oopsy), and so should probably get their own trigger.

If I wanted to stick my nose into this more and offer a potential solution, maybe this could have some helper function at the top like `function abilityWarn(name, id)` that returned a trigger object, in the style of `general.js`'s buffs?

Reply:
Mrrm, something like...

```
function abilityWarn(args) {
  if (!args.abilityId)
    console.error('Missing ability ' + JSON.stringify(args));
  return {
    id: args.id,
    netRegex: NetRegexes.ability({ abilityId: args.abilityId }),
    mistake: function(e, data, matches) {
       return { type: 'warn', blame: matches.target, text: matches.ability};
    },
}
```
I think that matches most oopsy triggers closely enough?

Reply:
Yeah, I think you may want to do abilityFull so you can check if `matches.flags.substr(-2) == '0E'` to cover the Empty Gaze too? Or handle that separately?

Reply:
No, that's a good point, something more like this then? Or am I overthinking?
```
function abilityWarn(args) {
  if (!args.abilityId)
    console.error('Missing ability ' + JSON.stringify(args));
  return {
    id: args.id,
    netRegex: NetRegexes.abilityFull({ abilityId: args.abilityId }),
    condition: function(e, data, matches) {
      if (typeof (args.condition) === undefined)
        return true;
      return args.condition;
    },
    mistake: function(e, data, matches) {
      return { type: 'warn', blame: matches.target, text: matches.ability };
    },
  };
}
```

Reply:
Don't all the failures for Bardam have `0E`? I guess I don't understand parameterizing the condition here.

Reply:
I'm not really sure myself now, four days on. I think maybe it was a misguided attempt at easier reusability for this helper function? But no, there's no reason not to just hardcode the flag as a condition.